File: exercises/java/src/main/java/com/adaptionsoft/games/trivia/runner/GameRunner.java

package com.adaptionsoft.games.trivia.runner;
import java.util.Random;

import com.adaptionsoft.games.uglytrivia.Game;
import com.adaptionsoft.games.uglytrivia.Player;


public class GameRunner {

	private static boolean notAWinner;

	public static void main(String[] args) {
		Game aGame = new Game();
		
		aGame.addPlayerToGame(new Player("Chet"));
		aGame.addPlayerToGame(new Player("Pat"));
		aGame.addPlayerToGame(new Player("Sue"));
		
		Random randomNumber = new Random(Integer.parseInt(args[0]));
	
		do {
			
			aGame.roll(randomNumber.nextInt(5) + 1);
			
			if (randomNumber.nextInt(9) == 7) {
				notAWinner = aGame.wrongAnswer();
			} else {
				notAWinner = aGame.checkAnswer();
			}
			
			
			
		} while (notAWinner);
		
	}
}

File: exercises/kotlin/src/main/kotlin/com/adaptionsoft/games/trivia/runner/GameRunner.kt
package com.adaptionsoft.games.trivia.runner

import com.adaptionsoft.games.uglytrivia.Game
import java.util.*

object GameRunner {
    fun main(args: Array<String>) {
        val aGame = Game()

        aGame.addPlayer("Chet")
        aGame.addPlayer("Pat")
        aGame.addPlayer("Sue")

        val rand = Random(args[0].toLong())

        do {

            aGame.roll(rand.nextInt(5) + 1)

            if (rand.nextInt(9) == 7) {
                GameRunner.notAWinner = aGame.wasIncorrectlyAnswered()
            } else {
                GameRunner.notAWinner = aGame.wasCorrectlyAnswered()
            }


        } while (GameRunner.notAWinner)

    }

    var notAWinner: Boolean = false
}

